sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2025,9 +2030,74 @@ class Analyzer(override val catalogManager: CatalogManager)
                   name, other.getClass.getCanonicalName)
               }
             }
-          case u @ UnresolvedFunction(funcId, arguments, isDistinct, filter, ignoreNulls) =>
+          case UnresolvedFunction(NonSessionCatalogAndIdentifier(v2Catalog, ident), arguments,
+            isDistinct, filter, _) if v2Catalog.isFunctionCatalog =>
```

Comment:
If the query refers to a function from catalog `cat`, but `cat` is not a `FunctionCatalog`, what error we should throw? `cat` doesn't support function or function not exist?

Reply:
This is a good point. We can check this in `LookupFunctions` and I think reporting that `cat` doesn't support function would be more informative in this case.